src/json.ts

```diff
@@ -0,0 +1,139 @@
+import deepEqual from 'fast-deep-equal';
+import { hasProperty } from './misc';
+
+/**
+ * Any JSON-compatible value.
+ */
+export type Json =
+  | null
+  | boolean
+  | number
+  | string
+  | Json[]
+  | { [prop: string]: Json };
+
+/**
+ * Type guard for {@link Json}.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is valid JSON.
+ */
+export function isValidJson(value: unknown): value is Json {
+  try {
+    return deepEqual(value, JSON.parse(JSON.stringify(value)));
+  } catch (_) {
+    return false;
+  }
+}
+
+/**
+ * A String specifying the version of the JSON-RPC protocol.
+ * MUST be exactly "2.0".
+ */
+export type JsonRpcVersion2 = '2.0';
+
+/**
+ * The string '2.0'.
+ */
+export const jsonrpc2 = '2.0' as const;
```

Comment:
TypeScript can infer the type here, so the `as const` assertion is not necessary. Maybe consider this:
```suggestion
export const jsonrpc2: JsonRpcVersion2 = '2.0';
```
That guarantees that the string value is the same as the type.

Reply:
Actually, I think using `typeof` is better in this case. Reduces duplication.

Reply:
You mean like `type JsonRpcVersion2 = typeof jsonrpc2`?

Reply:
Yes!